src/main/kotlin/it/unibo/alchemist/AlchemistExecutionOptions.kt

```diff
@@ -54,8 +71,42 @@ data class AlchemistExecutionOptions(
          * Default final time to be used if no final time is specified.
          * Defaults to [Double.POSITIVE_INFINITY].
          */
-        val defaultEndTime = Double.POSITIVE_INFINITY
+        const val defaultEndTime = Double.POSITIVE_INFINITY
+
+        /**
+         * Default engine mode to be used if not engine mode is specified.
+         * Defaults to [EngineMode.DETERMINISTIC].
+         */
+        val defaultEngineMode = EngineMode.DETERMINISTIC
 
         private val NO_OPTION = AlchemistExecutionOptions()
+
+        /**
+         * @property engineMode engine execution mode.
+         * @property outputReplayStrategy batch mode engine output replay strategy.
+         * @property batchSize fixed batch mode batch size.
+         * @property epsilon epsilon batch mode epsilon threshold.
+         */
+        data class EngineConfig(
```

Comment:
This looks like it has percolated from the implementation of the engine. It should not be there, defaults for the engine should be specified in the engine itself

Reply:
Removed, see general changes to Loader and Launcher creation